Wikipedia/Code/WMFCompassView.m

```diff
@@ -223,4 +223,21 @@ - (void)drawCompassLinesInContext:(CGContextRef)ctx
     }
 }
 
+#pragma mark - Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    double positive_angle = self.angleRadians >= 0 ? self.angleRadians : self.angleRadians + 2*3.14;
+    NSInteger clock_direction = lroundf(positive_angle / (3.14 / 6));
+    if (clock_direction == 0) {
+        clock_direction = 12;
+    }
+    NSString* label = MWLocalizedString(@"compass-direction", nil);
+    label = [label stringByReplacingOccurrencesOfString:@"$1" withString:[NSString localizedStringWithFormat:@"%@", @(clock_direction)]];
```

Comment:
> Regarding the first comment, isn't localizedStringWithFormat necessary for languages which use other characters for numerals, e.g. Japanese?

Oh, didn't consider that... we might need to do that in some other places as well.  For example, using the Swift REPL to print a localized string arabic numeral(?):

```
 8> NSString(format: "%@", locale:NSLocale(localeIdentifier:"ar"), NSNumber(int: 1))
$R6: NSString = "١"
  9> NSString(format: "%d", locale:NSLocale(localeIdentifier:"ar"), 1)
$R7: NSString = "١"
```